### PR TITLE
Fix ONNX node name sanitization and add ai.onnx.ml domain support

### LINKED ISSUE #2878: bug(onnx): model created using tf2onnx panics with non-valid Ident
**Describe the bug**
I have a equinox jax model I want to import into burn. I used `jax2tf` to get a tensorflow model, then `tf2onnx` to obtain the onnx file. When I try to import this into burn it panics with the message:

```
ERROR burn_import::logger: PANIC => panicked at /home/mrobins/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/burn-import-0.16.0/src/burn/ty.rs:123:19:
  "jax2tf_rhs_/pjit_silu_/Const_2:0" is not a valid Ident   
```

I'd imagine it is the ":" or "/" characters in the ident, which seem to be used to identify blocks and outputs in the model. Is this panic expected or a bug?

**To Reproduce**

You can get the onnx file here: https://github.com/martinjrobins/diffsol/raw/refs/heads/workspace/examples/neural-ode-weather-prediction/rhs.onnx

Then I read it in as per the onnx example: https://burn.dev/burn-book/import/onnx-model.html

**Expected behavior**

The model to import without panic

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser chrome
 - Version 22.04

**Additional context**
Using burn v0.16.0

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [x] Confirmed that `cargo run-checks` command has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

  - Fixes #2878 - ONNX import fails with panic on node names containing special characters

###  Changes

  Fixed ONNX import issues:

  1. Node name sanitization: Enhanced format_name function to sanitize node names with invalid identifier characters (:, /, etc.) by replacing them with underscores
  2. ONNX domain support: Added support for ai.onnx.ml domain in addition to standard ONNX domain

  This allows importing ONNX models generated by tf2onnx that contain special characters in node names.

### Testing

  - Added unit tests covering the problematic node name from the issue and various edge cases
  - Verified the original failing ONNX file now imports successfully
  - All existing tests pass

### CODE REVIEW

**Comment by antimora** [comment]
The name sanitization is happening in burn-import instead of onnx-ir. This is simpler for now and updates existing sanitization in place related. Added a todo to come back. #3208 PR attempted to fix in onnx-ir but added up introducing new issues because some names relied on old unsanitized names thus breaking code.

File: crates/burn-import/src/burn/ty.rs [hunk 1]
```diff
@@ -63,13 +63,31 @@ pub enum Type {
...
+                    }
+                })
+                .collect::<String>();
+
+            // Ensure the name starts with a letter or underscore
+            if sanitized.starts_with(|c: char| c.is_ascii_alphabetic() || c == '_') {
+                sanitized
+            } else {
+                format!("_{sanitized}")
+            }
```
Comment (laggui) [inline crates/burn-import/src/burn/ty.rs]: I think we can further simplify the general check with all these conditions, e.g.

```rust
pub fn format_name(name: &str) -> String {
    let mut result = String::with_capacity(name.len());
    // Sanitize the name by replacing invalid identifier characters with underscores
    for c in name.chars() {
        if c.is_ascii_alphanumeric() || c == '_' {
            result.push(c);
        } else {
            result.push('_');
        }
    }

    // Ensure the first character is valid to start an identifier
    if !result.starts_with(|c: char| c.is_ascii_alphabetic() || c == '_') {
        result = format!("_{result}");
    }
    result
}
```

mostly removes a duplicate check for the starting character

File: crates/onnx-ir/src/util.rs [hunk 2]
```diff
@@ -53,15 +53,21 @@ pub fn shape_config(curr: &Node) -> (usize, usize) {
...
+/// * If the domain is not supported
 pub fn check_opset_version(opset: &OperatorSetIdProto, min_version: i64) -> bool {
-    // For now, only empty domain (standard ONNX operators) is supported
-    if !opset.domain.is_empty() {
-        panic!("Only the standard ONNX domain is supported");
+    match opset.domain.as_str() {
+        // Standard ONNX operators
+        "" => opset.version >= min_version,
+        // ONNX ML operators - commonly used for traditional ML operators
+        "ai.onnx.ml" => opset.version >= 1, // ML operators are generally stable from version 1
```
Comment (laggui) [inline crates/onnx-ir/src/util.rs:62]: Adding this domain means we need to extend the ONNX op support list, no? We currently only list the basic ops in `SUPPORTED-ONNX-OPS.md`, not part of `ai.onnx.ml`.

**Review by laggui — COMMENTED** [review]
Fix looks good, just a minor comment.

What's the motivation behind extending support for `ai.onnx.ml` domain? We would need a separate list for [these ops](https://github.com/onnx/onnx/blob/main/docs/Operators-ml.md), which I'm not sure is desirable.

**Comment by antimora** [comment]
> Fix looks good, just a minor comment for the implementation 🙂
> 
> What's the motivation behind extending support for `ai.onnx.ml` domain? We would need a separate list for [these ops](https://github.com/onnx/onnx/blob/main/docs/Operators-ml.md), which I'm not sure is desirable.

Sorry I should have explained. TensorFlow adds this when exporting even though no ops are used. I have discovered this when testing an onnx file in #2878

File: crates/onnx-ir/src/util.rs (hunk 2 above)
Comment (antimora) [reply crates/onnx-ir/src/util.rs:62]: Maybe we should mention this since it's part of the ONNX spec.

File: crates/burn-import/SUPPORTED-ONNX-OPS.md [hunk 3]
```diff
@@ -206,6 +206,32 @@ functionality.
...
+| [Normalizer][210]                | ❌             | ❌           |
+| [OneHotEncoder][211]             | ❌             | ❌           |
+| [Scaler][212]                    | ❌             | ❌           |
+| [SVMClassifier][213]             | ❌             | ❌           |
+| [SVMRegressor][214]              | ❌             | ❌           |
+| [TreeEnsemble][215]              | ❌             | ❌           |
+| [TreeEnsembleClassifier][216]    | ❌             | ❌           |
+| [TreeEnsembleRegressor][217]     | ❌             | ❌           |
+| [ZipMap][218]                    | ❌             | ❌           |
+
```
Comment (laggui) [inline crates/burn-import/SUPPORTED-ONNX-OPS.md]: Not sure I would add the table. From the looks of these operators, I don't think we'll plan to support them any time soon 😅 so might be misleading.

We can accept the domain in the check for the reason you mentioned, but I think I would keep it at that.

**Review by laggui — COMMENTED** [review]
> Sorry I should have explained. TensorFlow adds this when exporting even though no ops are used.

Ahh ok I see, LGTM then!

But I think we should just accept the domain without listing the ops.

**Review by laggui — APPROVED** [review]
Thank you!
